### PR TITLE
Revised dGPU Driver Installation Process

### PR DESCRIPTION
**Pull request by liu-shaojun**
### 1. Why the change?

This submission reorganizes the installation process for dGPU drivers. For client GPUs (Intel® Arc™ A-series) on kernel 6.2 and above, upstream drivers are already supported, eliminating the need for out-of-tree drivers. For data center GPUs, including Intel® Data Center GPU Max Series and Intel® Data Center GPU Flex Series, the out-of-tree drivers are the only drivers validated by Intel.

Related Issue: https://github.com/analytics-zoo/nano/issues/1563
Related PR: https://github.com/intel-analytics/ipex-llm/pull/11728

### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: docs/mddocs/Quickstart/install_linux_gpu.md [hunk 1]
```diff
@@ -30,15 +30,10 @@ IPEX-LLM currently supports the Ubuntu 20.04 operating system and later, and sup
...
 
     ```bash
     sudo apt-get update
-    sudo apt-get -y install \
-        gawk \
-        dkms \
-        linux-headers-$(uname -r) \
-        libc6-dev
 
     sudo apt-get install -y gawk libc6-dev udev\
```
Comment (qiyuangong) [inline docs/mddocs/Quickstart/install_linux_gpu.md:38]: libc6-dev and gawk are also unncessary

File: docs/mddocs/Quickstart/install_linux_gpu.md [hunk 2]
```diff
@@ -78,15 +71,10 @@ IPEX-LLM currently supports the Ubuntu 20.04 operating system and later, and sup
...
```
Comment (qiyuangong) [inline docs/mddocs/Quickstart/install_linux_gpu.md:79]: The same

**Review by qiyuangong — APPROVED** [review]
Other, LGTM

**Comment by liu-shaojun** [comment]
After purging the out-of-tree driver and rebooting Arc11, the performance gap became significantly larger. Therefore, it is recommended to continue using the out-of-tree driver for testing. See performance gap in https://github.com/analytics-zoo/nano/issues/1563#issuecomment-2282981842

**Pull request closed without merging**